qiskit/quantum_info/operators/symplectic/clifford.py

```diff
@@ -110,109 +113,290 @@ def __array__(self, dtype=None):
+                    )
+                    num_qubits = data.shape[0] // 2
+                elif data.shape[0] + 1 == data.shape[1]:
+                    self.tableau = data
+                    num_qubits = data.shape[0] // 2
+                else:
+                    raise QiskitError("")
+            else:
+                n_paulis = len(data)
+                symp = self._from_label(data[0])
+                num_qubits = len(symp) // 2
+                tableau = np.zeros((n_paulis, len(symp)), dtype=bool)
+                tableau[0] = symp
+                for i in range(1, n_paulis):
+                    tableau[i] = self._from_label(data[i])
+                self.tableau = tableau
 
             # Validate table is a symplectic matrix
-            if validate and not Clifford._is_symplectic(self._table.array):
+            if validate and not Clifford._is_symplectic(self.symplectic_matrix):
                 raise QiskitError(
                     "Invalid Clifford. Input StabilizerTable is not a valid symplectic matrix."
                 )
 
         # Initialize BaseOperator
-        super().__init__(num_qubits=self._table.num_qubits)
+        super().__init__(num_qubits=num_qubits)
 
     @property
     def name(self):
         """Unique string identifier for operation type."""
         return "clifford"
 
-    @property
-    def num_qubits(self):
-        """Number of qubits."""
-        return self._table.num_qubits
-
     @property
     def num_clbits(self):
         """Number of classical bits."""
         return 0
 
     def __repr__(self):
-        return f"Clifford({repr(self.table)})"
+        return f"Clifford({repr(self.tableau)})"
 
     def __str__(self):
-        return "Clifford: Stabilizer = {}, Destabilizer = {}".format(
-            str(self.stabilizer.to_labels()), str(self.destabilizer.to_labels())
+        return (
+            f'Clifford: Stabilizer = {self.to_labels(mode="S")}, '
+            f'Destabilizer = {self.to_labels(mode="D")}'
         )
 
     def __eq__(self, other):
         """Check if two Clifford tables are equal"""
-        return super().__eq__(other) and self._table == other._table
+        return super().__eq__(other) and (self.tableau == other.tableau).all()
 
     # ---------------------------------------------------------------------
     # Attributes
     # ---------------------------------------------------------------------
+
+    # pylint: disable=bad-docstring-quotes
+
+    @deprecate_function(
+        "Indexing or iterating through a Clifford object directly is deprecated as of "
+        "Qiskit Terra 0.22.0 and will be removed no sooner than 3 months after the release date. "
+        "Instead, index or iterate through the Clifford.tableau attribute."
+    )
     def __getitem__(self, key):
         """Return a stabilizer Pauli row"""
-        return self._table.__getitem__(key)
+        return self.table.__getitem__(key)
```

Comment:
The `tableau` attribute is new in this release, so we shouldn't really be issuing a `DeprecationWarning` telling users to use it unless there's already been a previous release with `tableau` in it.